entity-management-definitions/src/main/java/eu/europeana/entitymanagement/normalization/EntityFieldsValidator.java

```diff
@@ -0,0 +1,299 @@
+		
+		List<Field> entityFields = new ArrayList<>();
+		EntityUtils.getAllFields(entityFields, entity.getClass());
+		
+		for (Field field : entityFields) {
+			try {			
+				Object fieldValue = entity.getFieldValue(field);
+				if (fieldValue==null) continue;
+				Class<?> fieldType = field.getType();
+				String fieldName = field.getName();
+				/*
+				 * the validation rules are implemented here
+				 */
+				//for the Text field type, check if it is a string and not a reference, and check the language codes
+				if (EntityFieldsTypes.valueOf(fieldName).getFieldType().equals("Text") || EntityFieldsTypes.valueOf(fieldName).getFieldType().equals("Keyword")) {
+					returnValue = validateTextOrKeywordField(context, field, fieldValue, returnValue);
+				}
+				//for the URI field type, check if it is a reference
+				else if (EntityFieldsTypes.valueOf(fieldName).getFieldType().equals("URI") || EntityFieldsTypes.valueOf(fieldName).getFieldType().equals("Date") || EntityFieldsTypes.valueOf(fieldName).getFieldType().equals("Email")) {
+					returnValue = validateEmailOrDateOrURIField(context, field, fieldValue, returnValue);
+				}				
+				//check the leading and trailing spaces from the String fields
+				else if (fieldType.isAssignableFrom(String.class)) {					
+					returnValue = validateStringField(context, field, fieldValue, returnValue);
+				}				
+			} catch (IllegalArgumentException e) {
+			    logger.warn("During the validation of the entity fields an illegal or inappropriate argument exception has happened.",e);
+			} catch (IllegalAccessException e) {
+			    logger.warn("During the validation of the entity fields an illegal access to some method or field has happened.",e);
+			} 
+		}
+		return returnValue;
+    }
+
+    boolean validateStringField(ConstraintValidatorContext context, Field field, Object fieldValue, boolean returnValue) throws IllegalAccessException {
+	if (!fieldValue.toString().trim().equals(fieldValue.toString())) {
+		addConstraint(returnValue, context, "The entity field: "+field.getName()+", contains leading and/or trailing spaces: "+fieldValue.toString()+".");
+		return false;
+	}
+	else {
+		return true;
+    }
+    }
+    
+    boolean validateEmailOrDateOrURIField(ConstraintValidatorContext context, Field field, Object fieldValue, boolean returnValue) {
+    	if (field.getType().isArray()) {
+			returnValue = validateEmailOrDateOrURIArrayField(context, field, fieldValue, returnValue);
+		}
+        else if (field.getType().isAssignableFrom(ArrayList.class)) {
+			returnValue = validateEmailOrDateOrURIListField(context, field, fieldValue, returnValue);
+		}
+        else if (field.getType().isAssignableFrom(String.class)){
+			returnValue = validateEmailOrDateOrURIStringField(context, field, fieldValue, returnValue);
+        }
+        return returnValue;
+    }
+    
+    boolean validateEmailOrDateOrURIArrayField(ConstraintValidatorContext context, Field field, Object fieldValue, boolean returnValue) {
```

Comment:
please separate the validation of Email, Date and URI, there is no field which has the type EmailDateOrURI

Reply:
Done.